gigascience/gigadb-website
# New feature/1569 merge authors modal should trap focus

# Pull request for issue: #1569

This is a pull request for the following functionalities:

* Modal related to merging authors in http://gigadb.gigasciencejournal.com:9170/adminAuthor/admin traps focus (tab key focus constrainted to modal content)
* On modal close, focus is returned to element that had focus before modal opens (usually button that opened the modal)
* Added button to open modal

## How to test?

* Navigate to http://gigadb.gigasciencejournal.com:9170/adminAuthor/update/id/14
* Click on "merge author..."
* Click on any merge button (new actions button)
* Use tab key to verify that focus cannot get out of modal
* Close modal and verify that focus returns to merge button that was clicked

## How have functionalities been implemented?

* Added a merge button that allows to open modal, so that keyboard only users can open merge modal, and there is a visual cue to open modal
* Conditionally display merge button
* Added some aria properties to the modal
* Fixed a11y issues within the modal
* Replaced merge authors link and close modal link by buttons
* Added JS code that traps focus in modal

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
